backend/python/trading_env/lib/python3.11/site-packages/click/_termui_impl.py

```
@@ -0,0 +1,788 @@
+
+else:
+    import termios
+    import tty
+
+    @contextlib.contextmanager
+    def raw_terminal() -> t.Iterator[int]:
+        f: t.Optional[t.TextIO]
+        fd: int
+
+        if not isatty(sys.stdin):
+            f = open("/dev/tty")
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)